### PR TITLE
Add Puppet8x support

### PR DESCRIPTION
**Pull request by flepoutre**

### CODE REVIEW

**Review by tuxmea — CHANGES_REQUESTED** [review]

File: manifests/init.pp [hunk 1]
```diff
@@ -3,21 +3,23 @@
...
-  $purge        = true,
+  String $files_owner  = 'root',
+  String $files_group  = 'root',
+  String $version      = present,
+  String $ruby_package = $augeas::params::ruby_pkg,
+  String $ruby_version = present,
+  String $lens_dir     = $augeas::params::lens_dir,
+  Boolean $purge       = true,
 ) inherits augeas::params {
-  if versioncmp($::puppetversion, '4.0.0') >= 0 {
```
Comment (bastelfreak) [inline manifests/init.pp:20]: I think we can remove the else block. We only support Puppet 7 and newer

File: manifests/lens.pp [hunk 2]
```diff
@@ -23,12 +28,12 @@
...
-  $lens_source  = undef,
-  $test_content = undef,
-  $test_source  = undef,
-  $stock_since  = false,
+  String $ensure                 = present,
+  Optional[String] $lens_content = undef,
+  Optional[String] $lens_source  = undef,
+  Optional[String] $test_content = undef,
+  Optional[String] $test_source  = undef,
+  Optional[String] $stock_since  = undef,
```
Comment (bastelfreak) [inline manifests/lens.pp:36]: I don't think we accept empty strings for any of those parameters? In that case please update it to `String[1]`

File: manifests/lens.pp [hunk 3]
```diff
@@ -58,7 +63,7 @@
     mode => '0644',
   }
 
-  if (!$stock_since or versioncmp(String($::augeasversion), $stock_since) < 0) {
+  if (!$stock_since or versioncmp($facts['augeasversion'], $stock_since) < 0) {
```
Comment (bastelfreak) [inline manifests/lens.pp:66]: that's a deprecated fact. please use 
```suggestion
  if (!$stock_since or versioncmp($facts['augeas']['version'], $stock_since) < 0) {
```

**Comment by jhoblitt** [comment]
Resolved by #99.

**Pull request closed without merging**